File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    pair_priorities = deque()
    for i, p in enumerate(priorities):
        pair_priorities.append((i, p))
        
    count = 1
    while pair_priorities:
        cur_i, cur_p = pair_priorities.popleft()
        
        re_append = False
        for next_i, next_p in pair_priorities:
            if next_p > cur_p:
                re_append = True
        
        if re_append:
            pair_priorities.append((cur_i, cur_p))
        elif cur_i == location:
            return count
        else:
            count += 1

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        var q = new LinkedList<int[]>(); // priority, index
        
        for (var i = 0; i < priorities.length; i++) {
            q.addLast(new int[]{priorities[i], i});
        }
        
        while (q.size() > 0) {
            var e = q.removeFirst();
            var priority = e[0];
            var index = e[1];
            
            var reAppend = false;
            for (var remain: q) {
                if (remain[0] > priority) {
                    reAppend = true;
                    break;
                }
            }
            
            if (reAppend) {
                q.addLast(e);
            }
            else {
                if (index == location) return answer;
                answer += 1;
            }
        }
        
        return answer;
    }
}